src/common/rand/rand.c

```diff
@@ -28,11 +30,19 @@ void OQS_randombytes_openssl(uint8_t *random_array, size_t bytes_to_read);
 // Use OpenSSL's RAND_bytes as the default PRNG
 static void (*oqs_randombytes_algorithm)(uint8_t *, size_t) = &OQS_randombytes_openssl;
 #else
+#if !defined(OQS_EMBEDDED_BUILD)
 static void (*oqs_randombytes_algorithm)(uint8_t *, size_t) = &OQS_randombytes_system;
+#else
+static void (*oqs_randombytes_algorithm)(uint8_t *, size_t) = NULL;
+#endif
 #endif
 OQS_API OQS_STATUS OQS_randombytes_switch_algorithm(const char *algorithm) {
 	if (0 == strcasecmp(OQS_RAND_alg_system, algorithm)) {
+#if !defined(OQS_EMBEDDED_BUILD)
 		oqs_randombytes_algorithm = &OQS_randombytes_system;
+#else
+		oqs_randombytes_algorithm = NULL;
```

Comment:
With the current implementation, you are right. 

Currently, the implementation of the `OQS_randombytes_system` function is completely disabled when `OQS_EMBEDDED_BUILD` is enabled. Hence, we cannot switch “back” to the system implementation using `OQS_randombytes_switch_algorithm`, as there is no implementation for it at all. 

As I saw it, currently  `OQS_randombytes_switch_algorithm` is only used within the test applications in the `tests` directory (which don't use `OQS_EMBEDDED_BUILD'). But I may miss some other important places.

One solution I can think of is to store the function pointer provided by `OQS_randombytes_custom_algorithm` internally in a second variable and use that as the “new” system implementation. So, when the user changed the RNG algorithm using `OQS_randombytes_switch_algorithm` and then changes it back to “system”, the method originally provided by `OQS_randombytes_custom_algorithm` is used again. This would make sure that `OQS_randombytes_switch_algorithm` works as intended all the time. However, this would permanently override the default system implementation (until reboot). 

Another solution could be to add an empty implementation of `OQS_randombytes_system` in case of an enabled `OQS_EMBEDDED_BUILD` causing a runtime error stating that one has to call `OQS_randombytes_switch_algorithm` providing a custom method. In this case, the user has to call `OQS_randombytes_custom_algorithm`  again after changing algorithms using `OQS_randombytes_switch_algorithm`. 

What do you think about these possibilities? Or do you have an idea for another solution?

Reply:
> I may miss some other important places.

Well, it's a public API so we can't know all places but have to assume (and support arbitrary) usage.

> What do you think about these possibilities? 

I very much like option 2: a) It only adds code in case "OQS_EMBEDDED_BUILD" is set, b) only impacts users calling `OQS_randombytes_switch_algorithm` with this option set (i.e., no present users) and c) adds a warning for the unwary. So would you please add this? Please also include a warning as per the above in the comment documentation of `OQS_randombytes_switch_algorithm`.

Reply:
> So would you please add this? Please also include a warning as per the above in the comment documentation of `OQS_randombytes_switch_algorithm`.

I updated the commits to incorporate option 2 and added the warning. This should indeed be the best solution without changes to existing users. 

I also did a small cleanup in `rand.c` regarding all the different `OQS_randombytes_system` implementations that are present. Hope that is ok.